projects/packages/forms/src/contact-form/class-contact-form-plugin.php

```diff
@@ -308,6 +307,55 @@ private static function register_contact_form_blocks() {
 		Contact_Form_Block::register_child_blocks();
 	}
 
+	/**
+	 * Build the CSS for the child layout.
+	 * This replicates part of the `wp_render_layout_support_flag` function from WordPress core.
+	 * That function doesn't work for form fields due to the way they're rendered as shortcodes,
+	 * while the core function will only generate and apply classnames for layout to html.
+	 *
+	 * @param array $layout - the block's `style.layout` attribute.
+	 *
+	 * @return string CSS for the child layout.
+	 */
+	private static function build_child_layout_css( $layout ) {
+		$css = '';
+
+		if ( ! $layout ) {
+			return $css;
+		}
+
+		$self_stretch = isset( $layout['selfStretch'] ) ? $layout['selfStretch'] : null;
+
+		if ( 'fixed' === $self_stretch && isset( $layout['flexSize'] ) ) {
+			$css .= 'flex-basis: ' . $layout['flexSize'] . ';';
+			$css .= 'box-sizing: border-box;';
+		} elseif ( 'fill' === $self_stretch ) {
+			$css .= 'flex-grow: 1;';
+		}
+
+		$column_start = isset( $layout['columnStart'] ) ? $layout['columnStart'] : null;
+		$column_span  = isset( $layout['columnSpan'] ) ? $layout['columnSpan'] : null;
+		if ( $column_start && $column_span ) {
+			$css .= "grid-column: $column_start / span $column_span;";
+		} elseif ( $column_start ) {
+			$css .= "grid-column: $column_start;";
+		} elseif ( $column_span ) {
+			$css .= "grid-column: span $column_span;";
+		}
+
+		$row_start = isset( $layout['rowStart'] ) ? $layout['rowStart'] : null;
+		$row_span  = isset( $layout['rowSpan'] ) ? $layout['rowSpan'] : null;
+		if ( $row_start && $row_span ) {
+			$css .= "grid-row: $row_start / span $row_span;";
+		} elseif ( $row_start ) {
+			$css .= "grid-row: $row_start;";
+		} elseif ( $row_span ) {
+			$css .= "grid-row: span $row_span;";
+		}
+
+		return $css;
+	}
```

Comment:
It's a bit unfortunate this is needed, but the positive of doing it this way vs. other things I've tried is that it's very easy to understand and super easy to fix any bugs. It doesn't rely on any tricks and isn't much code.

The other options were all too brittle.